sdk/containerapps/Azure.ResourceManager.AppContainers/src/Custom/Models/ContainerAppVnetConfiguration.Serialization.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.ResourceManager.AppContainers.Models
+{
+    [CodeGenSerialization(nameof(InfrastructureSubnetId), DeserializationValueHook = nameof(DeserializeInfrastructureSubnetIdValue))]
+    public partial class ContainerAppVnetConfiguration : IUtf8JsonSerializable, IJsonModel<ContainerAppVnetConfiguration>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void DeserializeInfrastructureSubnetIdValue(JsonProperty property, ref ResourceIdentifier infrastructureSubnetId)
+        {
+            if (property.Value.ValueKind != JsonValueKind.Null && !string.IsNullOrEmpty(property.Value.GetString()))
+            {
+                infrastructureSubnetId = new ResourceIdentifier(property.Value.GetString());
+            }
+            else
+            {
+                throw new ArgumentNullException("infrastructureSubnetId cannot be null or empty.");
```

Comment:
Don't throw exception here otherwise it's still failed when retrieve empty subnet id from service.
And here please try using `models-to-treat-empty-string-as-null`

Reply:
Resolved